### ISSUE TITLE
test检测不出框

### ISSUE THREAD

**Issue by jiaerwang0328** [qqwweee/keras-yolo3#413]
WARNING:tensorflow:From D:\anaconda\anacondainstall\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
model_data/yolo.h5 model, anchors, and classes loaded.
Input image filename:dog.jpg
(416, 416, 3)
Found 0 boxes for img
2.9271458189999997
Input image filename:

**Comment by jjjenny1008** [comment]
你的显示框的score设置过高，把score值调整低一些就可以显示了

**Comment by jiaerwang0328** [comment]
class YOLO(object):
    _defaults = {
        "model_path": 'model_data/yolo.h5',
        "anchors_path": 'model_data/yolo_anchors.txt',
        "classes_path": 'model_data/coco_classes.txt',
        "score" : 0.3,
        "iou" : 0.45,
        "model_image_size" : (416, 416),
        "gpu_num" : 1,
    }
 默认是这样的，我尝试把score改为0.1 还是检测不出来，

**Comment by lcltopismine** [comment]
试试看print一下self.boxes, self.scores, self.classes，确定一下是不是score改了，因为default是0.6，另外iou的结果也影响检测，但是iou是和training有关的，所以不清楚是不是train的时候没收敛。
另外你可以测一下看看是不是你的test image都检测不出来，有些图像比较难检测

**Comment by liumgg** [comment]
请问这个问题你解决了么，我出现同样的问题，iou和score调小都没有检测出框，用训练数据检测也没有很好的识别

**Comment by IrisDinge** [comment]
> 请问这个问题你解决了么，我出现同样的问题，iou和score调小都没有检测出框，用训练数据检测也没有很好的识别

请问你解决了吗？

**Comment by IrisDinge** [comment]
> 试试看print一下self.boxes, self.scores, self.classes，确定一下是不是score改了，因为default是0.6，另外iou的结果也影响检测，但是iou是和training有关的，所以不清楚是不是train的时候没收敛。
> 另外你可以测一下看看是不是你的test image都检测不出来，有些图像比较难检测

你好，我也遇到了相同的问题想请教你。训练DOTA数据集，loss9.645但是测试的时候，只有score为0的时候才能检测出来框，但是置信度全是0，相当于测试失败。 我不是很明白，为什么训练结果不算差但是测试结果非常差，甚至用训练的数据做测试也得不到结果。请问您有什么建议或者意见吗？

**Comment by a793719595** [comment]
check whether the training data is consistent with the test data in structure (both RGB or BGR)

**Comment by xm0629** [comment]
> 请问这个问题你解决了么，我出现同样的问题，iou和score调小都没有检测出框，用训练数据检测也没有很好的识别

**Comment by a793719595** [comment]
qw大佬的视频检测程序有点问题 图像检测没问题 你要是图像出不来框就是训练出问题了




------------------ 原始邮件 ------------------
发件人: "Ming Xu"<notifications@github.com>;
发送时间: 2019年8月28日(星期三) 下午3:24
收件人: "qqwweee/keras-yolo3"<keras-yolo3@noreply.github.com>;
抄送: "山大王悦"<793719595@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [qqwweee/keras-yolo3] test检测不出框 (#413)



 
请问这个问题你解决了么，我出现同样的问题，iou和score调小都没有检测出框，用训练数据检测也没有很好的识别

**Comment by Zhang-O** [comment]
how did you get your model? I train my model using darknet with a cfg . Then I convert it from weights to h5. And I detect nothing. You may check your train cfg and the cfg used by convert.py and ocr.name. When I do this , all is ok

**Comment by sunyhaime** [comment]
我也遇到了同样的问题，请问楼主解决了吗？

**Comment by dydxdt** [comment]
I use yolo3 to test vkitti dataset(only has car and truck class , car takes most part of the labels). And the object confidence and class score is quite low: e-2 or e-3. The mAP is also quite low, even though I set the conf_threshold, nms_threshold, iou_threshold to a very small value. In fact, I don't think it is a right choice to do so, but using original normal value, AP is 0.  Does anyone else can give me some advice?  The original test dataset is from COCO and mAP can reach 0.51. 
Really hope for help!
